src/Language/Fortran/Parser/Free/Lexer.x

```diff
@@ -1093,22 +1103,29 @@ advance move position =
     _line = posLine position
     _absl = posAbsoluteOffset position
 
--- Handle pragmas that begin with #
-lexHash :: LexAction (Maybe Token)
-lexHash = do
-  ai <- getAlex
-  m <- getMatch
-  case words (drop 1 m) of
+processLinePragma :: String -> AlexInput -> AlexInput
+processLinePragma m ai =
+  case dropWhile ((`elem` ["#", "line", "#line"]) . map toLower) (words m) of
```

Comment:
Won't this happily process lines like `# line line #line # line 40 "file.f"`? Which it probably should complain about.